First steps in write Asserts for the test web site.

latajacysmok opened the pull request.

latajacysmok committed:
First steps in write Asserts.

whyyy commented on Learning_test_automation/Spotkanie8_Zadania/SeleniumCsharp/SeleniumCsharp/SeleniumCsharp.csproj:
```diff
@@ -0,0 +1,30 @@
+<Project Sdk="Microsoft.NET.Sdk">
```
Moznaby rozbic projekt `SeleniumCsharp`, bo teraz jest to projekt testowy pod NUnita + page objecty + selenium

latajacysmok replied:
W sensie rozbić na dwa projekty?

whyyy replied:
tak

whyyy commented on Learning_test_automation/Spotkanie8_Zadania/SeleniumCsharp/SeleniumCsharp/UnitTest1.cs:
```diff
@@ -0,0 +1,195 @@
+
+namespace SeleniumCsharp
+{
+    public class Tests
```
Zmienilbym na dokladniejsza nazwe tu i tak samo nazwe pliku, testow moze byc wiecej, dla roznych page objectow i wtedy zacznie sie balagan

latajacysmok replied:
hmmm... zmienić zmieniłem ale nie jestem pewny.

whyyy commented on Learning_test_automation/Spotkanie8_Zadania/SeleniumCsharp/SeleniumCsharp/UnitTest1.cs:
```diff
@@ -0,0 +1,195 @@
+{
+    public class Tests
+    {
+        String automationintesting_url = "https://automationintesting.online/";
```
Nazwa i konwencja nazewnictwa do poprawy

latajacysmok replied:
ok

whyyy commented on Learning_test_automation/Spotkanie8_Zadania/SeleniumCsharp/SeleniumCsharp/UnitTest1.cs:
```diff
@@ -0,0 +1,195 @@
+            driver = new ChromeDriver();
+            driver.Manage().Window.Maximize();
+        }
+
```
Za duzo wciec w niektorych testach, przykladowy screenshot:
![image](https://github.com/latajacysmok/Road_to_Csharp_developer/assets/11249681/93c05448-1a26-450a-a3a2-fae1c6396218)

latajacysmok replied:
racyja.

whyyy commented on Learning_test_automation/Spotkanie8_Zadania/SeleniumCsharp/SeleniumCsharp/UnitTest1.cs:
```diff
@@ -0,0 +1,195 @@
+        }
+
+        [Test]
+        public void EnterToWeb()
```
Nazwa tego testu i przynajmniej niektorych pozostalych moglaby byc dokladniejsza, tak, zeby sama nazwa dawala informacje co jest w zakresie danego testu, jaki rezultat oczekujemy, itp, itd.
W artykule, ktory omawialismy odnosnie selenium, jest kilka przykladow (nieidealnych, ale dokladniejszych)
![image](https://github.com/latajacysmok/Road_to_Csharp_developer/assets/11249681/7b6aec9d-80be-45ff-8b47-769fc5fe3e51)

Sa tez konwencje nazewnictwa testow, przyklady tu:
https://dzone.com/articles/7-popular-unit-test-naming 

konwencje 5-7 sa chyba najpopularniejsze, a 7ma bazuje na BDD i jest podobna do gherkina ;)

latajacysmok replied:
Na razie spróbowałem 5.

whyyy commented on Learning_test_automation/Spotkanie8_Zadania/SeleniumCsharp/SeleniumCsharp/UnitTest1.cs:
```diff
@@ -0,0 +1,195 @@
+                home.driver.Navigate().GoToUrl(automationintesting_url);
+
+                var headingBookRoomSection = new WebDriverWait(driver, TimeSpan.FromSeconds(3))
+                                .Until(drv => home.HeadingBookRoomSection.Text);
```
zamiast tworzenia zmiennej `drv` w metodzie `Until` mozna uzyc discard, wiecej o tym tu:
https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/functional/discards

latajacysmok replied:
o! Super, nie trzeba wymyślać nazwy 🗡️

whyyy commented on Learning_test_automation/Spotkanie8_Zadania/SeleniumCsharp/SeleniumCsharp/UnitTest1.cs:
```diff
@@ -0,0 +1,195 @@
+                            .Until(drv => home.ContentBookRoomArticle.Text);
+
+            //Assert.True(contentBookRoomArticle.StartsWith("Aenean") && contentBookRoomArticle.EndsWith("ante."));
+            contentBookRoomArticle.Should().StartWith("Aenean").And.EndWith("ante.").And.Contain("accumsan").And.HaveLength(146);
```
👍

latajacysmok replied:
ja woll!

whyyy commented on Learning_test_automation/Spotkanie8_Zadania/SeleniumCsharp/SeleniumCsharp/UnitTest1.cs:
```diff
@@ -0,0 +1,195 @@
+
+            bookDateCancelButton.Click();
+
+            Action calendarNotDispaly = () => driver.FindElement(By.XPath(@"//div[@class = 'rbc-calendar']"));
```
Literowka w nazwie

latajacysmok replied:
Ach te literówki xD

whyyy commented on Learning_test_automation/Spotkanie8_Zadania/SeleniumCsharp/SeleniumCsharp/UnitTest1.cs:
```diff
@@ -0,0 +1,195 @@
+
+            Action calendarNotDispaly = () => driver.FindElement(By.XPath(@"//div[@class = 'rbc-calendar']"));
+
+            calendarNotDispaly.Should().Throw<NoSuchElementException>("The element should not be present.");
```
👍

latajacysmok replied:
:D

whyyy reviewed (changes requested):
Calkiem fajnie to wyglada 👍 
Pare uwag i jeszcze jedno pytanie.
Ten `Hotel.png` plik jest uzywany gdzies, potrzebny?

latajacysmok committed:
Add new tests

Change the architecture of the project and add some new tests and make some method.

latajacysmok committed:
I add Test in Spec Flow convention.

latajacysmok committed:
We can Try.

latajacysmok committed:
More tests, wrap some selenium methods etc.
